formats/json-tests/jvmTest/src/kotlinx/serialization/json/JsonChunkedDecoderTest.kt

```diff
@@ -0,0 +1,146 @@
+    }
+}
+
+@Serializable(with = LargeStringSerializer::class)
+data class LargeStringData(val largeString: String)
+
+@Serializable
+data class ClassWithLargeStringDataField(val largeStringField: LargeStringData)
+
+
+object LargeStringSerializer : KSerializer<LargeStringData> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("LargeStringContent", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): LargeStringData {
+        require(decoder is ChunkedDecoder) { "Only chunked decoder supported" }
+
+        val outStringBuilder = StringBuilder()
+
+        decoder.decodeStringChunked { chunk ->
+            outStringBuilder.append(chunk)
+        }
+        return LargeStringData(outStringBuilder.toString())
+    }
+
+    override fun serialize(encoder: Encoder, value: LargeStringData) {
+        encoder.encodeString(value.largeString)
+    }
+}
+
+
+
+class JsonChunkedDecoderTest:JsonTestBase() {
+
+    @Test
+    fun decodeBase64String() {
+        val sourceObject = ClassWithBinaryDataField(LargeBinaryData(Random.nextBytes(16 * 1024))) // After encoding to Base64 will be larger than 16k (JsonLexer#BATCH_SIZE)
+        val serializedObject = Json.encodeToString(sourceObject)
+
+        JsonTestingMode.values().forEach { mode ->
+            if (mode == JsonTestingMode.TREE) {
+                assertFailsWithMessage<IllegalArgumentException>( "Only chunked decoder supported", "Shouldn't decode JSON in TREE mode") {
+                    Json.decodeFromString<ClassWithBinaryDataField>(serializedObject, mode)
+                }
+            } else {
+                val deserializedObject = Json.decodeFromString<ClassWithBinaryDataField>(serializedObject, mode)
+                assertEquals(sourceObject.binaryField, deserializedObject.binaryField)
+            }
+        }
+    }
+
+    @Test
+    fun decodePlainLenientString() {
+        val sourceObject = ClassWithLargeStringDataField(LargeStringData("abcdef"))
+        val serializedObject = "{\"largeStringField\": abcdef }"
+        val lenientJson = Json {
+            isLenient = true
+        }
+        JsonTestingMode.values().forEach { mode ->
+            if (mode == JsonTestingMode.TREE) {
+                assertFailsWithMessage<IllegalArgumentException>( "Only chunked decoder supported", "Shouldn't decode JSON in TREE mode") {
+                    Json.decodeFromString<ClassWithLargeStringDataField>(serializedObject, mode)
+                }
+            } else {
+                val deserializedObject = lenientJson.decodeFromString<ClassWithLargeStringDataField>(serializedObject, mode)
+                assertEquals(sourceObject.largeStringField, deserializedObject.largeStringField)
+            }
+        }
+    }
+
+    @Test
+    fun decodePlainString() {
+        val sourceObject = ClassWithLargeStringDataField(LargeStringData("abc\"def")) // After encoding to Base64 will be larger than 16k (JsonLexer#BATCH_SIZE)
```

Comment:
Comment `After encoding to Base64 will be larger than 16k (JsonLexer#BATCH_SIZE)` is kinda misleading here — I think it was copied by accident?

Reply:
Definitely